src/newton_solver.cpp

```diff
@@ -410,38 +368,150 @@ int NewtonSolverIterative::prepareLinearSystem(int ntry, int nnewt) {
      * @param[in] ntry integer newton_try integer start number of Newton solver
      * (1 or 2)
      * @param[in] nnewt integer number of current Newton step
-     * @return stats integer flag indicating success of the method
      */
 
     newton_try = ntry;
     i_newton = nnewt;
     if (nnewt == -1) {
-        return AMICI_ERROR_NEWTON_LINSOLVER;
-    } else {
-        return AMICI_SUCCESS;
+        throw NewtonFailure("No setup necessary");
```

Comment:
Here, the error message should be
throw NewtonFailure("Linear solver SPBCG does not support sensitivity computation for steady state problems");